src/components/CityComponent.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useEffect, useState } from 'react';
+import { StyleSheet, View, Text, ImageBackground } from 'react-native';
+import { utcToZonedTime, format } from 'date-fns-tz';
+import { isSameMinute } from 'date-fns';
+import { ru } from 'date-fns/locale';
+import colors from '@styles/colors';
+import { Signs } from '@utils/constants';
+
+export interface ICityComponent {
+    location: string;
+    timeZone: string;
+}
+
+const initialCityData = {
+    weatherType: 'Облачно',
+    temperature: 12,
+    maxTemperature: 12,
+    minTemperature: -3,
+    image: require('@assets/icons/backgroundRain.png'),
+};
+
+export const CityComponent: React.FC<ICityComponent> = props => {
+    const [cityData, setCityData] = useState(initialCityData);
+    const [currentTime, setCurrentDate] = useState(new Date());
+    useEffect(() => {
+        const intervalId = setInterval(() => {
+            isSameMinute(currentTime, new Date()) ? setCurrentDate(currentTime) : setCurrentDate(new Date());
```

Comment:
You shouldn't cause a re-render if nothing changes. Also your function is executed every millisecond, it is 1000 times a second, so you are causing re-render 1000 times a second when only need to re-render once a minute. Change interval to 1000 to call every second, don't call setCurrentDate(currenttime)